src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,51 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_PASSWORD_SIZE = 6;
+    private static final int MIN_AGE_FOR_REGISTER = 18;
+    private static final int MAX_AGE = 125;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkNull(user);
+        checkCorrectAge(user);
+        return userAdd(user);
+    }
+
+    private User userAdd(User user) {
+        User findUser = storageDao.get(user.getLogin());
+        if (findUser != null) {
+            throw new RuntimeException("User login is already taken");
+        }
+        if (user.getPassword().length() < MIN_PASSWORD_SIZE) {
+            throw new RuntimeException("User password must be is at least 6 characters");
+        }
```

Comment:
Make this in separate method, like you do with checkCorrectAge.  checkCorrectPassword(), maybe)